pkg/ccl/backupccl/restore_planning.go

```diff
@@ -1788,11 +1767,45 @@ func doRestorePlan(
 	newDBName string,
 	endTime hlc.Timestamp,
 	resultsCh chan<- tree.Datums,
+	subdir string,
 ) error {
 	if len(from) < 1 || len(from[0]) < 1 {
 		return errors.New("invalid base backup specified")
 	}
 
+	var defaultIncFrom []string
+	var err error
+	if subdir == "" {
+		if defaultIncFrom, err = appendPaths(from[0][:], DefaultIncrementalsSubdir); err != nil {
+			return err
+		}
+	} else {
+		if strings.EqualFold(subdir, "LATEST") {
+			// set subdir to content of latest file
+			latest, err := readLatestFile(ctx, from[0][0], p.ExecCfg().DistSQLSrv.ExternalStorageFromURI, p.User())
+			if err != nil {
+				return err
+			}
+			subdir = latest
+		}
+		if len(from) != 1 {
+			return errors.Errorf("RESTORE FROM ... IN can only by used against a single collection path (per-locality)")
+		}
+
+		if defaultIncFrom, err = appendPaths(from[0][:], path.Join(DefaultIncrementalsSubdir, subdir)); err != nil {
```

Comment:
nit: might be clearer to move this switch statement below len(incfrom), and then add the following comment above the first first `appendPaths` call: append subdirectory to full backup path first, then to candidate incremental storage paths.

again, I wonder if it's possible to only use `incFrom`, instead of both `incFrom` and `defaultIncFrom`

Reply:
I think that `if len` check isn't needed, so I'll remove it - not what you were asking about, but it's just a little bit clearer :)

Reply:
to continue my ramble from my comment above, what if we moved all the logic from `if subdir == "" {} else{}` into it's own function along the lines of `resolveBackupDests(from, subdir, incFrom) (resolvedFrom [][]string, resolvedIncFrom []string, incrementalbackupSource enum )` . The function could even return a backupLocStruct instead.

The one complication here is we'll need to open a store to figure out if we should use the new or old default. But I think this is fine.

We could then use this function in ShowBackups.

Happy to chat through this idea.